nullaway/src/test/java/com/uber/nullaway/NullAwayCustomLibraryModelsTests.java

```diff
@@ -227,4 +227,38 @@ public void libraryModelsAndSelectiveSkippingViaCommandLineOptions2() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void libraryModelsAndOverridingFieldNullability() {
```

Comment:
I realized we need to support / test one more scenario.  If code writes a `@Nullable` value to a library field modeled as `@Nullable` there should be no error.  Not sure that will work right now.  Can we test?

Reply:
@msridhar The test fails. The impact of not having this support is that Annotator under rare circumstances might reject a fix tree that there is a direct nullable assignment in downstream dependencies to one of the fields in the tree which also the overall impact is less than 0. Please let me know if we want this support or you think we should just skip for now. Overall, I think for consistency we should have this support.

One way to fix that is that we need to add a custom handling for checking if the field is annotated as `@Nullable` [here](https://github.com/uber/NullAway/blob/5fbee1ff8682762cd2f291070491fe3e476787cf/nullaway/src/main/java/com/uber/nullaway/NullAway.java#L489). That might require adding a new method to handlers (e.g. `hasNullableAnnotation`, please let me know if we can use existing methods). In that case please let me know if I should add this support in this PR or have a followup PR after this one lands. Thank you.

Reply:
I agree that it is likely rare that we would want to model a library field as `@Nullable` _and_ clients of the library are writing `@Nullable` values into the field.  But I worry that the incomplete support will bite us down the road, and even that we'll run into some unexpected case with the annotator.  So I say let's just implement the support in this PR.

I agree with your strategy of adding a new `Handler` method; we can declare it `boolean onOverrideFieldNullability(Symbol fieldSymbol)` to keep things simple for now.  It only gets called if the field is not annotated `@Nullable`, and if it returns `true`, the field should be treated as `@Nullable` anyway.  Let me know if it doesn't make sense.  Thanks!

Reply:
@msridhar Sure, sounds good. But I also found the method below in handlers API:
`Nullness onOverrideMethodReturnNullability(Symbol.MethodSymbol methodSymbol,
      VisitorState state,
      boolean isAnnotated,
      Nullness returnNullness)`

I thought it might be more consistent with other methods and its name (overriding the passed value as arg), if we have the method declared as:
`Nullness onOverrideFieldNullability(Symbol field, Nullness fieldNullness);`

We can also comply with the contract you mentioned for calling the method in the method body. Please let me know if I should use a different declaration. Thank you.

8347141ce8ddc3323b3cc5231759d881031b5c12

Reply:
I suggested the contract above for performance and to limit unnecessary generality.  With method return nullability I think we need more complex overriding functionality, which is why it has the rather complex behavior and signature.  With fields, I personally think it's ok to just say if any handler makes a field nullable, then it's nullable.  What do you think?  I personally prefer the simpler approach and then refactoring later if we need it.

Reply:
@msridhar Sure, makes sense. We really don't need flexibility here. Please find the updated version 4d8187131ec57e117903d302a8d7d096b21b250b

Reply:
Thanks this looks great now.